apis/s3/src/test/java/org/jclouds/s3/filters/RequestAuthorizeSignatureV4Test.java

```diff
@@ -83,6 +84,11 @@
          + "SignedHeaders=content-length;content-type;host;x-amz-content-sha256;x-amz-date;x-amz-storage-class, "
          + "Signature=090f1bb1db984221ae1a20c5d12a82820a0d74b4be85f20daa1431604f41df08";
 
+   private static final String LIST_BUCKET_RESULT = "AWS4-HMAC-SHA256 "
+         + "Credential=AKIAPAEBI3QI4EXAMPLE/20150203/cn-north-1/s3/aws4_request, "
+         + "SignedHeaders=host;x-amz-content-sha256;x-amz-date, "
+         + "Signature=6cc5d0758e2599be7cb172fd57cefab2828201a2b4d372972a83dc304de93958";
```

Comment:
Merely based on running the test input against rawQuery. I did not attempt to consult the AWS documentation and manually construct the signature base as per [these instructions](https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html).

The key to the fix is that `Queries.parseKeyValueFromStringToDecodedMap` is used to URL-decode the parameters of the query string after splitting on `&`, and we then reëncode in `getCanonicalizedQueryString`, so using `URI.getQuery()` (which decodes the value) not only double-decodes, it actually loses information. For example,

```
URI u = new URI("https://x.com/?p1=a&p2=b%26c%3Dd&p3=e");
System.out.println(u.getRawQuery());
System.out.println(u.getQuery());
```

will print

```
p1=a&p2=b%26c%3Dd&p3=e
p1=a&p2=b&c=d&p3=e
```